react/week3/todolistapp/src/components/TodoListInputs.jsx

```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+import DatePicker from 'react-date-picker';
+import { format } from 'date-fns';
+
+export default function TodoListInputs(props) {
+    const {addTodo} = props;
+    const [description, setDescription] = useState("");
+    const [deadline, setDeadline] = useState();
+    const [addMessage, setAddMessage]= useState("Please add Todo Description and Correct time to enable Add Todo Button");
+    let disabled = true;
```

Comment:
I would not store the state of disabled like this. There is a few ways you could do this. I see you are checking for description and deadline and if they dont exist then disabled is true and you set disabled to false if the date criteria is met. You could actually just derive the logic from state and do two variables one for the description and deadline exist and another variable that checks for the date and then use these variable in disabled. It don't have to be like this, you could also use state, it is up to you, but i would not do it like this.